src/workerd/api/gpu/gpu.c++

```diff
@@ -21,7 +22,17 @@ void initialize() {
   dawnProcSetProcs(&dawn::native::GetProcs());
 }
 
-GPU::GPU() {
+GPU::GPU(CompatibilityFlags::Reader flags) {
+  // is this a durable object?
+  KJ_IF_MAYBE (actor, IoContext::current().getActor()) {
+    JSG_REQUIRE(actor->getPersistent() != nullptr, TypeError,
+                "webgpu api is only available in Durable Objects (no storage)");
+  } else {
+    JSG_FAIL_REQUIRE(TypeError, "webgpu api is only available in Durable Objects (no actor)");
```

Comment:
Since this is a user-facing error and we don't use the "actor" terminology that much with public facing stuff, I'd remove the `(no actor)` part of the message.